server/apis/v1/handler.go

```diff
@@ -1180,6 +1189,143 @@ func (h *handler) GetMonoVertexHealth(c *gin.Context) {
 	c.JSON(http.StatusOK, NewNumaflowAPIResponse(nil, response))
 }
 
+func formatArrayLabels(arrLabels []interface{}) string {
+	var labelSelectors []string
+	for _, value := range arrLabels {
+		labelSelectors = append(labelSelectors, fmt.Sprintf(`%s`, value))
+	}
+	return strings.Join(labelSelectors, ",")
+}
+
+func formatMapLabels(mapLabels map[string]interface{}) string {
+	var labelSelectors []string
+	for key, value := range mapLabels {
+		labelSelectors = append(labelSelectors, fmt.Sprintf(`%s="%s"`, key, value))
+	}
+	return strings.Join(labelSelectors, ",")
+}
+
+func (h *handler) GetMetricData(c *gin.Context) {
+
+	if h.prometheusClient == nil || h.prometheusClient.Client == nil {
+		h.respondWithError(c, "Prometheus client not set.")
+		return
+	}
+
+	var index int = -1
+	var fieldsMap = make(map[string]any)
+
+	var (
+		requestBody     map[string]any
```

Comment:
Shall me abstract the input data into a struct? `map[string]any` means no restriction.